macos/Sources/Features/Terminal/TerminalManager.swift

```diff
@@ -17,7 +17,13 @@ class TerminalManager {
     var focusedSurface: Ghostty.SurfaceView? { mainWindow?.controller.focusedSurface }
 
     /// The set of windows we currently have.
-    var windows: [Window] = []
+    private(set) var windows: [Window] = [] {
+        didSet {
```

Comment:
Yeah let's remove all this machinery since we aren't using it. 

I think using Combine in the future is a better approach than this anyways.